### DISCUSSION TITLE
Disable the `unicorn/prefer-string-slice` rule

### DISCUSSION THREAD

**Discussion by leroykorterink** [mediamonks/eslint-config#234] [General]
There is a proposal to disable the `unicorn/prefer-string-slice` rule.

> `unicorn/prefer-string-slice`: `String.substr` and `String.substring` work in slightly different ways, and sometimes we need to use one or the other.
> \- @edgardz

**Comment by leroykorterink** [comment]
We should keep the `prefer-string-slice` rule because `String#substr()` is deprecated and `String#substring()` is a lesser known option to slice a string. It's better to use `String#slice()` as it has clearer behavior that has a consistent `Array` counterpart.

The rule is auto-fixable so it's low effort to fix. Also, I don't want to mess with recommended configurations (disable rules that are recommended) when it's not necessary, because it may increase maintenance work.

Please see [the documentation](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-string-slice.md
) for the rule

**Comment by ThaNarie** [comment]
`slice` is the improved version over `substring`, so there is no reason to keep using `substring`, unless you rely on the "magic" behaviour that it swaps the start/end indices in case they are the wrong way around.

```ts
// does not support negative indices, so you have to use `foo.length - 5`
foo.substring(indexStart, indexEnd);

// does support negative indices to count backwards, so you can just pass `-1` or `-5`
foo.slice(beginIndex, endIndex);
```

`substr` is officially deprecated, so should not be used under any circumstances, as it might be breaking in the future.

If you really need the **length** feature of `substr` – `substr(start, length)` – you can do `slice(start, start+length)` instead. Or not, and it will autofix it for you :)

**Comment by edgardz** [comment]
ok then...
